### PR TITLE
[Bug] [API] relation delete api bug

### PR DESCRIPTION
**Pull request by brave-lee**
close by #8146

### CODE REVIEW

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessTaskRelationController.java [hunk 1]
```diff
@@ -101,46 +101,6 @@ public Result createProcessTaskRelation(@ApiIgnore @RequestAttribute(value = Con
...
-        @ApiImplicitParam(name = "projectCode", value = "PROJECT_CODE", required = true, type = "Long"),
-        @ApiImplicitParam(name = "processDefinitionCode", value = "PROCESS_DEFINITION_CODE", required = true, type = "Long"),
-        @ApiImplicitParam(name = "targetProcessDefinitionCode", value = "TARGET_PROCESS_DEFINITION_CODE", required = true, type = "Long"),
-        @ApiImplicitParam(name = "taskCode", value = "TASK_CODE", required = true, type = "Long")
-    })
-    @PostMapping(value = "/move")
-    @ResponseStatus(HttpStatus.OK)
-    @ApiException(MOVE_PROCESS_TASK_RELATION_ERROR)
-    @AccessLogAnnotation(ignoreRequestArgs = "loginUser")
-    public Result moveTaskProcessRelation(@ApiIgnore @RequestAttribute(value = Constants.SESSION_USER) User loginUser,
```
Comment (caishunfeng) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessTaskRelationController.java:125]: Is it OK to delete the controller function directly?

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessTaskRelationServiceImpl.java [hunk 2]
```diff
@@ -372,6 +259,21 @@ private void updateProcessDefiniteVersion(User loginUser, Map<String, Object> re
         return result;
     }
 
+    private void updateRelation(User loginUser, Map<String, Object> result, ProcessDefinition processDefinition,
+                                List<ProcessTaskRelation> processTaskRelationList) {
+        updateProcessDefiniteVersion(loginUser, result, processDefinition);
+        List<ProcessTaskRelationLog> relationLogs = processTaskRelationList.stream().map(ProcessTaskRelationLog::new).collect(Collectors.toList());
+        int insertResult = processService.saveTaskRelation(loginUser, processDefinition.getProjectCode(), processDefinition.getCode(),
+            processDefinition.getVersion(), relationLogs, Lists.newArrayList(), Boolean.TRUE);
+        if (insertResult == Constants.EXIT_CODE_SUCCESS) {
```
Comment (caishunfeng) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessTaskRelationServiceImpl.java:268]: It's better to use another const,  but not `EXIT_CODE_SUCCESS`

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessTaskRelationServiceImpl.java [hunk 3]
```diff
@@ -555,47 +474,44 @@ private void updateProcessDefiniteVersion(User loginUser, Map<String, Object> re
...
-        int insert = processTaskRelationLogMapper.insert(processTaskRelationLog);
-        if ((update & insert) == 0) {
-            putMsg(result, Status.DELETE_EDGE_ERROR);
-            throw new ServiceException(Status.DELETE_EDGE_ERROR);
+        } else {
+            ProcessTaskRelation processTaskRelation = taskRelationMap.get(postTaskCode).get(0);
+            processTaskRelationList.remove(processTaskRelation);
+            processTaskRelation.setPreTaskVersion(0);
+            processTaskRelation.setPreTaskCode(0L);
+            processTaskRelationList.add(processTaskRelation);
```
Comment (caishunfeng) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessTaskRelationServiceImpl.java:512]: why to add a empty preTask here?

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessTaskRelationController.java (hunk 1 above)
Comment (brave-lee) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessTaskRelationController.java:125]: There is a logic bug that cannot be bypassed. Providing this method will lead to inconsistent workflow versions

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessTaskRelationServiceImpl.java (hunk 2 above)
Comment (brave-lee) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessTaskRelationServiceImpl.java:268]: This comes from processservice, which is unified with others

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessTaskRelationServiceImpl.java (hunk 3 above)
Comment (brave-lee) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessTaskRelationServiceImpl.java:512]: This method is to delete the dependency between two tasks. When the node is not clustered, just set the prefix to 0, which means to delete a line

**Pull request merged**